promql/parser/parse_test.go

```diff
@@ -1852,6 +1852,48 @@ var testExpr = []struct {
 			},
 		},
 	},
+	{
+		// Specifying __name__ twice inside the brackets is ok
```

Comment:
Nit: End sentence with a period (same below).

And maybe use "braces" instead of "brackets" to avoid ambiguity? ("Braces" is also used a lot already in this file, but "brackets" never.)